### PR TITLE
Kubernetes configuration for bitcoin and electrumx

### PR DESCRIPTION
**Pull request by nkuba**
This PR adds Kubernetes manifests of bitcoin node and electrumx server.

The configuration is based on the V1's config: https://github.com/keep-network/tbtc/tree/main/infrastructure/kube/keep-test with some improvements.

For V1 we were running bcoin, now we switch to bitcoind, due to https://github.com/bcoin-org/bcoin/issues/1153

Bitcoind and Electrum servers are running in dedicated kubernetes namespaces:
- for keep-test: `bitcoin-testnet`
- for keep-prd: `bitcoin`

We started with one replica for each server and after sync was done, we created snapshots that were used for PVC creation for other replicas.

Closes: https://github.com/keep-network/keep-core/issues/3590

### CODE REVIEW

File: infrastructure/kube/keep-prd/bitcoin/bitcoind/bitcoind-data-bitcoind-1-pvc.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: bitcoind-data-bitcoind-1
```
Comment (pdyraga) [inline infrastructure/kube/keep-prd/bitcoin/bitcoind/bitcoind-data-bitcoind-1-pvc.yaml:5]: I am looking at the list of PVCs in GCP and I see `bitcoind-data-bitcoind-1` and `bitcoind-data-bitcoind-0`. To confirm my understanding: we are running two replicas of bitcoind, each with its own PVC. Then, why don't we have `bitcoind-data-bitcoind-0-pvc.yaml` file here as well?

File: infrastructure/kube/keep-prd/bitcoin/bitcoind/bitcoind-volumesnapshot.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: bitcoind-snapshot
+spec:
+  volumeSnapshotClassName: bitcoind
+  source:
+    persistentVolumeClaimName: bitcoind-data-bitcoind-0
```
Comment (pdyraga) [inline infrastructure/kube/keep-prd/bitcoin/bitcoind/bitcoind-volumesnapshot.yaml:8]: Why is it just for `-0`? From the PR, I understand that we first synced one instance, then used the snapshots to sync the second instance faster. Long-term, don't we want to have snapshots for both PVCs? If `-0` gets corrupted for whatever reason, we could restore it from `-1`.

File: infrastructure/kube/keep-prd/bitcoin/electrumx/electrumx-volumesnapshot.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: electrumx-snapshot
+spec:
+  volumeSnapshotClassName: electrumx
+  source:
+    persistentVolumeClaimName: electrumx-data-electrumx-0
```
Comment (pdyraga) [inline infrastructure/kube/keep-prd/bitcoin/electrumx/electrumx-volumesnapshot.yaml:8]: Same question about generating snapshots as for [bitcoind](https://github.com/keep-network/keep-core/pull/3573/files#r1223912755). Don't we need `-1` snapshots as well?

File: infrastructure/kube/templates/bitcoin/bcoin/.env.template [hunk 4]
```diff
@@ -0,0 +1,2 @@
+http-api-key=
+bcoin-url=
```
Comment (pdyraga) [inline infrastructure/kube/templates/bitcoin/bcoin/.env.template:2]: Is it used anywhere?

File: infrastructure/kube/keep-prd/bitcoin/bitcoind/bitcoind-volumesnapshot.yaml (hunk 2 above)
Comment (nkuba) [reply infrastructure/kube/keep-prd/bitcoin/bitcoind/bitcoind-volumesnapshot.yaml:8]: We needed snapshot from `bitcoind-data-bitcoind-0` to spin up the second replica. There was no point of creating a snapshot of `bitcoind-data-bitcoind-1`, as they would be pretty close. After some time I plan to take a new snapshot and repeat it periodically.

File: infrastructure/kube/templates/bitcoin/bcoin/.env.template (hunk 4 above)
Comment (nkuba) [reply infrastructure/kube/templates/bitcoin/bcoin/.env.template:2]: We should remove the whole `bcoin` setup.

File: infrastructure/kube/keep-prd/bitcoin/electrumx/electrumx-volumesnapshot.yaml (hunk 3 above)
Comment (pdyraga) [reply infrastructure/kube/keep-prd/bitcoin/electrumx/electrumx-volumesnapshot.yaml:8]: Answered [here](https://github.com/keep-network/keep-core/pull/3573#discussion_r1224417079).

File: infrastructure/kube/templates/bitcoin/bcoin/.env.template (hunk 4 above)
Comment (pdyraga) [reply infrastructure/kube/templates/bitcoin/bcoin/.env.template:2]: Ack. I'll then hold on until this is addressed.

Comment (nkuba) [reply infrastructure/kube/templates/bitcoin/bcoin/.env.template]: 7f082d1910f70fd87e71fe1ee26d14aeaeaa5273

**Review by pdyraga — APPROVED** [review]

**Pull request merged**